### PR TITLE
[vcpkg baseline][cairo] Add option atomic-ops and disable it

### PR DESCRIPTION
**Pull request by JackBoosY**
cairo will automatically detect `libatomic-ops` and enable this feature when discovered. But using it in cairo has some issues.
I have opened an issue upstream: https://gitlab.freedesktop.org/cairo/cairo/-/issues/554
Since cairo uses auto-detection instead of options, we can't add dependencies explicitly, so I added a feature and disabled it until upstream fixes this.

Replace #23995.

### CODE REVIEW

**Review by LilyWangLL — APPROVED** [review]

File: ports/cairo/add-option-atomic-ops.patch [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ 
+-if cc.links(files('meson-cc-tests/atomic-ops-cxx11.c'), name: 'Atomic ops: cxx11')
+-  conf.set('HAVE_CXX11_ATOMIC_PRIMITIVES', 1)
+-elif cc.links(files('meson-cc-tests/atomic-ops-gcc-legacy.c'), name: 'Atomic ops: gcc legacy')
+-  conf.set('HAVE_GCC_LEGACY_ATOMICS', 1)
+-elif cc.has_header('atomic_ops.h')
+-  conf.set('HAVE_LIB_ATOMIC_OPS', 1)
+-elif cc.has_header('libkern/OSAtomic.h')
+-  conf.set('HAVE_OS_ATOMIC_OPS', 1)
++if get_option('atomic-ops').enabled()
```
Comment (strega-nil-ms) [inline ports/cairo/add-option-atomic-ops.patch:17]:
```suggestion
+if false
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
I also think we should just remove the check for `atomic_ops.h`, so that it will still check for `atomic-ops-cxx11` and `atomic-ops-gcc-legacy` and `libkern/OSAtomic.h` (OSX built-in atomics)

**Comment by JackBoosY** [comment]
@strega-nil-ms Done.

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**